### PR TITLE
use attention mask parameter in flux attention

### LINKED ISSUE #10025: attention mask for transformer Flux
### Describe the bug

Is it possible to get back the `attention_mask` argument in the flux attention processor 

```
hidden_states = F.scaled_dot_product_attention(query, key, value, dropout_p=0.0, is_causal=False,attn_mask=attention_mask)
```

https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/attention_processor.py#L1910

in order to tweak things a bit ? otherwise the argument  `attention_mask`  is unused.

Thanks a lot 

### Reproduction

pip install diffusers

### Logs

_No response_

### System Info

Ubuntu

### Who can help?

@yiyixuxu @sayakpaul @DN6 @asomoza

### PR DESCRIPTION
**Pull request by christopher5106**
Unused params `attention_mask` in transformer flux attention processors is useful to reweight prompt or image prompt.

Fixes #10025 

@yiyixuxu @sayakpaul

### CODE REVIEW

**Comment by bghira** [comment]
this is incomplete. more changes are required to make use of the text encoder attn masking since it's being combined with the image tokens.

**Comment by bonlime** [comment]
Thumbs up to this request. Diffusers is not only a set of pipelines, but also a library of use-full components, even if currently there is no diffusers pipeline that uses attention masking for flux, it doesn't mean it shouldn't be passed. I personally have a flux pipeline that requires passing attention mask and currently had to copy-paste existing attn processor only to start using this argument

**Comment by bghira** [comment]
i have code to do attn masking in flux and would like it to have feature parity with what Diffusers will do. hence this is incomplete.there is more than just passing args in using generic parameter names. you can use image_attention_mask and encoder_attention_mask and fill with ones if encoder mask is apllied butnimage mask isnt.

however i think more importantly there is a fundamental misapplication of attention masking happening here, because it is not for re-weighting inputs.

**Comment by bonlime** [comment]
This has been closed by #10122

**Comment by bghira** [comment]
i'll just keep mentioning that it's an incomplete fix that has been merged and that this is not yet resolved
